ivy/functional/frontends/numpy/indexing_routines/generating_index_arrays.py

```diff
@@ -30,9 +30,10 @@ def unravel_index(indices, shape, order="C"):
     return tuple(ret)
 
 
-@outputs_to_numpy_arrays
+@to_ivy_arrays_and_back
 def diag_indices(n, ndim=2):
-    idx = ivy.arange(n, dtype=int)
+    idx = ivy.arange(n)
+    idx = idx.astype("int64")
     return (idx,) * ndim
```

Comment:
you have to typecast the output before returning, we should not type cast the inputs in this case you have again typcasted the input.

```
res = (idx,) * ndim
res = res.astype("int64")
return res
```